contrib/pax_storage/src/cpp/access/pax_access_handle.cc

```diff
@@ -0,0 +1,1217 @@
+                                            SampleScanState *scanstate) {
+  CBDB_TRY();
+  {
+    auto desc = PaxScanDesc::ToDesc(scan);
+    return desc->ScanSampleNextBlock(scanstate);
+  }
+  CBDB_CATCH_DEFAULT();
+  CBDB_FINALLY({});
+  CBDB_END_TRY();
+  pg_unreachable();
+}
+
+bool CCPaxAccessMethod::ScanSampleNextTuple(TableScanDesc scan,
+                                            SampleScanState *scanstate,
+                                            TupleTableSlot *slot) {
+  CBDB_TRY();
+  {
+    auto desc = PaxScanDesc::ToDesc(scan);
+    return desc->ScanSampleNextTuple(scanstate, slot);
+  }
+  CBDB_CATCH_DEFAULT();
+  CBDB_FINALLY({});
+  CBDB_END_TRY();
+  pg_unreachable();
+}
+
+void CCPaxAccessMethod::MultiInsert(Relation relation, TupleTableSlot **slots,
+                                    int ntuples, CommandId cid, int options,
+                                    BulkInsertState bistate) {
+  CBDB_TRY();
+  {
+    MemoryContext old_ctx;
+    Assert(cbdb::pax_memory_context);
+
+    old_ctx = MemoryContextSwitchTo(cbdb::pax_memory_context);
+    CPaxInserter::MultiInsert(relation, slots, ntuples, cid, options, bistate);
+    MemoryContextSwitchTo(old_ctx);
+
+    pgstat_count_heap_insert(relation, ntuples);
+  }
+  CBDB_CATCH_DEFAULT();
+  CBDB_FINALLY({
+      // FIXME: destroy CPaxInserter?
+  });
+  CBDB_END_TRY();
+}
+
+void CCPaxAccessMethod::FinishBulkInsert(Relation relation, int options) {
+  // Implement Pax dml cleanup for case "create table xxx1 as select * from
+  // xxx2", which would not call ExtDmlFini callback function and relies on
+  // FinishBulkInsert callback function to cleanup its dml state.
+  CBDB_TRY();
+  {
+    // no need switch memory context
+    // cause it just call dml finish
+    pax::CPaxInserter::FinishBulkInsert(relation, options);
+  }
+  CBDB_CATCH_DEFAULT();
+  CBDB_FINALLY({
+      // FIXME: destroy CPaxInserter?
+  });
+  CBDB_END_TRY();
+}
+
+void CCPaxAccessMethod::ExtDmlInit(Relation rel, CmdType operation) {
+  if (!RELATION_IS_PAX(rel)) return;
+
+  CBDB_TRY();
+  { pax::CPaxDmlStateLocal::Instance()->InitDmlState(rel, operation); }
+  CBDB_CATCH_DEFAULT();
+  CBDB_FINALLY({});
+  CBDB_END_TRY();
+}
+
+void CCPaxAccessMethod::ExtDmlFini(Relation rel, CmdType operation) {
+  if (!RELATION_IS_PAX(rel)) return;
+
+  CBDB_TRY();
+  { pax::CPaxDmlStateLocal::Instance()->FinishDmlState(rel, operation); }
+  CBDB_CATCH_DEFAULT();
+  CBDB_FINALLY({});
+  CBDB_END_TRY();
+}
+
+}  // namespace pax
+// END of C++ implementation
+
+// access methods that are implemented in C
+namespace paxc {
+const TupleTableSlotOps *PaxAccessMethod::SlotCallbacks(
+    Relation /*rel*/) noexcept {
+  return &TTSOpsVirtual;
+}
+
+uint32 PaxAccessMethod::ScanFlags(Relation relation) {
+  uint32 flags = 0;
+#ifdef VEC_BUILD
+  flags |= SCAN_SUPPORT_VECTORIZATION | SCAN_SUPPORT_COLUMN_ORIENTED_SCAN;
+#else
+  flags |= SCAN_SUPPORT_COLUMN_ORIENTED_SCAN;
```

Comment:
add support `SCAN_SUPPORT_RUNTIME_FILTER`, I will push qual to pax as we done in cloud.

Reply:
I don't think that is a good idea to make PAX support runtime filter in current PR?

Reply:
ok